arcgis-ios-sdk-samples/Utility network/Create load report/CreateLoadReportViewController.swift

```diff
@@ -295,40 +284,42 @@ extension CreateLoadReportViewController: UITableViewDelegate, UITableViewDataSo
                 usingComparator: { ($0 as! AGSCodedValue).name.compare(($1 as! AGSCodedValue).name) }
             )
             excludedPhases.insert(phase, at: insertionIndex)
-            let section = Section.excluded.rawValue
-            tableView.insertRows(at: [IndexPath(row: insertionIndex, section: section)], with: .automatic)
+            // Remove the summary.
+            summaries.removeValue(forKey: phase)
+            tableView.insertRows(at: [IndexPath(row: insertionIndex, section: .excluded)], with: .automatic)
         case .insert:
-            tableView.deleteRows(at: [indexPath], with: .automatic)
             let phase = excludedPhases[indexPath.row]
             excludedPhases.remove(at: indexPath.row)
-            includedPhases.append(phase)
-            let section = Section.included.rawValue
-            let newRow = tableView.numberOfRows(inSection: section)
-            tableView.insertRows(at: [IndexPath(row: newRow, section: section)], with: .automatic)
+            let insertionIndex = (includedPhases as NSArray).index(
```

Comment:
According to Friday's peer review, the included phases would be better to be sorted alphabetically.

This part is pretty sad - how I wish I can create a reference to a Swift array! 😿 
Basically it is the same code for 2 arrays which order is decided by the editing style. Looking for some suggestions.

Reply:
> Basically it is the same code for 2 arrays

So write a helper function! It could just be nested under `tableView(_:commit:forRowAt:)`:

```swift
func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
    func index(forInserting phase: AGSCodedValue, into array: [AGSCodedValue]) -> Int {
        (array as NSArray).index(
            of: phase,
            inSortedRange: NSRange(array.indices),
            options: [.insertionIndex],
            usingComparator: { ($0 as! AGSCodedValue).name.compare(($1 as! AGSCodedValue).name) }
        )
    }
    
    ...
    excludedPhases.insert(phase, at: index(forInserting: phase, into: excludedPhases))
    ...
    includedPhases.insert(phase, at: index(forInserting: phase, into: includedPhases))
    ...
}
```